src/backend/cdb/motion/ic_udpifc.c

```diff
@@ -1167,6 +1169,105 @@ resetRxThreadError()
 }
 
 
+/*
+ * isIPv6Available
+ * 		Check whether a valid IPv6 address exists in the system.
+ *
+ * 		We want to have both loopback IPv6 address, which will show if IPv6 was
+ * 		disabled earlier, and some non-link-local address currently available
+ * 		for our use.
+ */
+static bool
+isIPv6Available(void)
+{
+	bool		hasIpv6Addr = false;
+	bool		hasIpv6Loopback = false;
+	int			errnoSave = errno;
+
+	int			errorErrnoSave = 0;
+	char	   *errorCause = NULL;
+
+	struct ifaddrs	*ifaddr = NULL;
+	struct ifaddrs	*ifaddrFirst = NULL;
+
+	if (getifaddrs(&ifaddrFirst) == -1)
+	{
+		errorCause = "getifaddrs()";
+		goto end;
+	}
+
+	for (ifaddr = ifaddrFirst; ifaddr != NULL; ifaddr = ifaddr->ifa_next)
+	{
+		if (ifaddr->ifa_addr != NULL &&
```

Comment:
may be use not condition continue here and below to get rid of too nested?

Reply:
I replaced extra nesting level for link-local/wildcard check with `continue`. If we also remove it here, we'll be unable to use `const` on address pointers.